dask/dataframe/io/parquet/arrow.py

```diff
@@ -1643,15 +1668,19 @@ def _read_table(
             # We only need to do this if we are applying filters to
             # columns that were not already filtered by "partition".
             if (partitions and partition_keys is None) or (
-                partitioning and _need_fragments(filters, partition_keys)
+                partitioning
+                and (
+                    not isinstance(partitioning, (str, list))
```

Comment:
Sorry, I missed this the first time. Why do we need to add `not isinstance(partitioning, (str, list))`?

Reply:
This is actually a really good question.  It turns out that we don't actually populate `ds.dictionaries` when the user passes in a custom partitioning scheme, and so we get `partitions == []` and need to create a `Dataset` object here to properly detect the partitioned column values.

The `(str, list)` line is really just checking if the user is asking for a custom `partitioning` scheme. Having str or list inputs means that we are not specifying a schema, and so we should have the usual categorical-column information.

I will admit that there is a bit of weirdness here that I don't really know how to address: There is really no way for a user to get categorical columns when they specify a custom schema, because they would need to use the `PartitioningFactory.discover` method (which doesn't play well with repeated execution or serialization).